### PR TITLE
Fix planning conflict e2e test errors

### PR DESCRIPTION
**Pull request by ysf-simsoft**
wdio e2e test were failing randomly. I have continuously run the tests on BrowserStack in order to observe the problem. After detailed investigation it turns out that `createReportWithPlanningConflict` tests were suffering from timeout because this particular suit took 4-5 minutes (on BrowserStack) to run on average. Particularly `"Should create second draft report with conflicts"` spec was intermittently taking more than 90 seconds to run. I made changes to element queries in order to reduce run time and increased the timeout duration for test runner. As a result timeout related failures _almost_ disappeared.

image

As can be seen in the screenshot above out of 1315 spec (~77 runs for whole suit) runs only 1 failure happened. And that particular failure is also happened due to a timeout but as can be seen in the raw webdriver logs in the following screenshot and BrowserStack logs on the latter secreenshot, webdriver waits for more than 60 seconds between two consecutive requests for querying an element. I couldn't figure out the reason for that as it is almost impossible to reproduce and debug any further.

image

image

Consequnetly `createReportWithPlanningConflict` suit's run time has decreased to ~3-3.5 minutes and `"Should create second draft report with conflicts"` specs run time has decreased by ~20 seconds. Even though there was an intermittent test failure, this PR will significantly reduce the timeout related issues, hopefully.

I also had to tweak `editLocations` and `editPositions` specs because they were also suffering from an `unexpected alert open` error on BrowserStack.

UPDATE:
I have merged these changes into my local [keycloak-integration branch](https://github.com/NCI-Agency/anet/tree/keycloak-integration) and wdio-tests have been running on BrowserStack for almost 24 hours. All wdio-tests ran more than 100 times without any failures. Last two sessions in the following screenshot doesn't have any failures. (Both sessions are timed out by BrowserStack btw :slightly_smiling_face: hence the yellow warning icons).

image

Closes #

#### User changes
-

#### Super User changes
-

#### Admin changes
-

#### System admin changes
- [ ] anet.yml or anet-dictionary.yml needs change
- [ ] db needs migration
- [ ] documentation has changed
- [ ] graphql schema has changed

### Checklist
  - [ ] Described the user behavior in PR body
  - [ ] Referenced/updated all related issues
  - [ ] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
  - [ ] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
  - [ ] Added and/or updated unit tests
  - [x] Added and/or updated e2e tests
  - [ ] Added and/or updated data migrations
  - [ ] Updated documentation
  - [ ] Resolved all build errors and warnings
  - [ ] Opened debt issues for anything not resolved here

### CODE REVIEW

File: client/tests/webdriver/pages/report/showReport.page.js [hunk 1]
```diff
@@ -81,10 +81,11 @@ class ShowReport extends Page {
   }
 
   waitForShowReportToLoad() {
-    if (!this.reportStatus.isDisplayed()) {
-      this.reportStatus.waitForExist()
-      this.reportStatus.waitForDisplayed()
-    }
+    browser.waitUntil(() =>
+      /^.*\/reports\/[a-z0-9-]{36}/.test(browser.getUrl())
```
Comment (gjvoosten) [inline client/tests/webdriver/pages/report/showReport.page.js:85]: This also matches the Edit page URL. Perhaps hook the regex at the end with a `$`?

File: client/config/wdio.config.js [hunk 2]
```diff
@@ -140,7 +140,7 @@ const config = {
   mochaOpts: {
     ui: "bdd",
     compilers: ["js:@babel/register"],
-    timeout: 90000
+    timeout: 180000
```
Comment (gjvoosten) [inline client/config/wdio.config.js:143]: Is this change necessary or was it just there for the tests?

**Review by gjvoosten — CHANGES_REQUESTED** [review]

File: client/tests/webdriver/pages/report/showReport.page.js (hunk 1 above)
Comment (ysf-simsoft) [reply client/tests/webdriver/pages/report/showReport.page.js:85]: It is supposed to match the Edit page URL, as well. On `keycloak-integration` branch I noticed that there is an intermediate URL change when navigating between pages. This change aims to handle that gracefully.

File: client/config/wdio.config.js (hunk 2 above)
Comment (ysf-simsoft) [reply client/config/wdio.config.js:143]: It is necessary as some tests, especially `"Should create second draft report with conflicts"`, sometimes take more than 120 seconds. I think 180 seconds will keep us on the safe side.

**Review by gjvoosten — APPROVED** [review]
